examples/picorv32_ram/picorv32_ram.py

```diff
@@ -0,0 +1,60 @@
+
+import importlib
+import os
+import siliconcompiler
+
+def build_top():
+    # Core settings.
+    design = 'picorv32_top'
+    target = 'skywater130_demo'
+    die_w = 1000
+    die_h = 1200
+
+    # Create Chip object.
+    chip = siliconcompiler.Chip(design)
+
+    # Set default Skywater130 PDK / standard cell lib / flow.
+    target_module = importlib.import_module(f'targets.{target}')
+    chip.use(target_module)
+
+    # Set design source files.
+    rootdir = os.path.dirname(__file__)
+    chip.input(os.path.join(rootdir, f"{design}.v"))
+    chip.input(os.path.join(rootdir, "picorv32.v"))
+    chip.input(os.path.join(rootdir, "sky130_sram_2k.bb.v"))
+    chip.input(os.path.join(rootdir, f"{design}.sdc"))
+
+    # Optional: Relax linting and/or silence each task's output in the terminal.
+    chip.set('option', 'relax', True)
+    chip.set('option', 'quiet', True)
+
+    # Set die outline and core area.
+    chip.set('constraint', 'outline', [(0,0), (die_w, die_h)])
+    chip.set('constraint', 'corearea', [(10,10), (die_w-10, die_h-10)])
+
+    # Setup SRAM macro library.
+    from sram import sky130_sram_2k
+    chip.use(sky130_sram_2k)
+    chip.add('asic', 'macrolib', 'sky130_sram_2k')
+
+    # Increase macro place halo to reduce global routing congestion on SRAM macro.
+    chip.set('tool', 'openroad', 'task', 'route', 'var', 'route', '0', 'grt_allow_congestion', 'true')
+    chip.set('tool', 'openroad', 'task', 'floorplan', 'var', 'floorplan', '0', 'macro_place_halo', '40')
+    chip.set('tool', 'openroad', 'task', 'floorplan', 'var', 'floorplan', '0', 'macro_place_channel', '40')
```

Comment:
I recall these macro_place arguments only affects the auto macro placer. In this example the SRAM is manually placed so they are possibly irrelevant?

Reply:
Yep, after some more experimenting, I think the only `var` that we need is `grt_macro_extension` set to `0`.

I'm not completely sure what that routing parameter is supposed to do, but it may be revisiting whether we should set it to a positive value by default. I think it might set routing blockage halos around each macro, maybe in case of pins that extend beyond the block's boundary?

Reply:
Grt_macro_extention causes the global router to route to a "fake" pin that is offset from the macro by that number of gcells. It's okay to have it be 0, but generally it needs to match the technology since it helps the detail router. If detail route finishes (which it does) then leaving it at 0 is okay